libc-test/build.rs

```diff
@@ -1866,3 +1867,67 @@ fn test_dragonflybsd(target: &str) {
 
     cfg.generate("../src/lib.rs", "main.rs");
 }
+
+fn test_wasi(target: &str) {
+    assert!(target.contains("wasi"));
+
+    let mut cfg = ctest::TestGenerator::new();
+    cfg.define("_GNU_SOURCE", None);
+
+    headers! { cfg:
+        "errno.h",
+        "fcntl.h",
+        "limits.h",
+        "locale.h",
+        "malloc.h",
+        "stddef.h",
+        "stdint.h",
+        "stdio.h",
+        "stdlib.h",
+        "sys/stat.h",
+        "sys/types.h",
+        "time.h",
+        "unistd.h",
+        "wasi/core.h",
+        "wasi/libc.h",
+        "wchar.h",
+    }
+
+    cfg.type_name(move |ty, is_struct, is_union| match ty {
+        "FILE" => ty.to_string(),
+        t if is_union => format!("union {}", t),
+        t if t.starts_with("__wasi") && t.ends_with("_u") => format!("union {}", t),
+        t if t.starts_with("__wasi") && is_struct => format!("struct {}", t),
+        t if t.ends_with("_t") => t.to_string(),
+        t if is_struct => format!("struct {}", t),
+        t => t.to_string(),
+    });
+
+    // This is an opaque struct but we go through shenanigans to define values
+    // for it
+    cfg.skip_struct(move |ty| ty == "__clockid");
```

Comment:
I'm just gonna go ahead an exclude `__clockid` because it's sufficiently "weird enough" and not entirely necessary yet that I think it's fine to defer it to later